src/index.spec.ts

```diff
@@ -1,67 +1,19 @@
+/**
+ * @jest-environment puppeteer
+ */
 import * as http from 'http';
 import puppeteer from 'puppeteer';
 import startServer from '../server';
 
-class SinglePage {
-    public readonly page: puppeteer.Page;
-    public messages: {
-        warning: string[]
-        log: string[]
-        info: string[]
-        debug: string[]
-        error: string[]
-    };
-    constructor(page: puppeteer.Page) {
-        this.page = page;
-        this.messages = {
-            debug: [],
-            error: [],
-            info: [],
-            log: [],
-            warning: [],
-        };
-        // collect browser console messages to be checked later
-        this.page.on('console', (consoleMessage) => {
-            const type = consoleMessage.type();
-            const text = consoleMessage.text();
-
-            switch (type) {
-                case 'debug':
-                case 'error':
-                case 'info':
-                case 'log':
-                case 'warning':
-                    this.messages[type].push(text);
-                    break;
-                default:
-                    throw new Error(`BUG: Unsupported console type: '${type}'`);
-            }
-        });
-
-        this.page.on('pageerror', async(e) => {
-            throw e;
-        });
-    }
-}
-
-async function startBrowser(browser: puppeteer.Browser, url: string) {
-
-    const page = await browser.newPage();
-
-    const wrapper = new SinglePage(page);
-    await wrapper.page.goto(url);
-
-    // Wait until React has started
-    await wrapper.page.waitForSelector('#root > h1');
-    return wrapper;
-}
+// jest-puppeteer will expose the `page` and `browser` globals to Jest tests.
+// declare const browser: puppeteer.Browser
+declare const page: puppeteer.Page;
 
 describe('Browser sanity tests', () => {
 
     let devServer: http.Server | null = null;
     let devServerPort = 0;
-    let browser: puppeteer.Browser | null = null;
-    let wrapper: SinglePage | null = null;
+    let consoleMessages: Array<{type: 'debug' | 'error' | 'info' | 'log' | 'warning', message: string}> = [];
 
     beforeAll(async() => {
         await startServer({
```

Comment:
we should specify custom globalSetup and globalTeardown hooks to start and stop the dev server